ragen/env/search/env.py

```
@@ -0,0 +1,550 @@
+from ragen.utils import all_seed
+import datasets
+
+
+class SearchEnv(BaseLanguageBasedEnv):
+    """Multi-turn search environment following Search-R1's pattern.
+    
+    Actions: LLM generates <search>query</search> or <answer>response</answer>
+    Environment: Executes search calls and appends results as <information> blocks
+    """
+
+    def __init__(self, config: Optional[SearchEnvConfig] = None, sys_config=None) -> None:
```

Ruff (0.14.10)

Unused method argument: `sys_config`

(ARG002)